### PR TITLE
Multipart Support for PUT method

### LINKED ISSUE #1262: Multipart not supported with put() method.
Currently rest-assured doesn't support Multipart File with `PUT` method.  
Following Exception is thrown when we try to use `multipart` method with `put()` : `java.lang.IllegalArgumentException: Currently multi-part file data uploading only works for POST`

Version : 3.0.3

### PR DESCRIPTION
**Pull request by anki2189**
This PR fixes #1262

### CODE REVIEW

**Comment by anki2189** [comment]
@johanhaleby : Please review the changes :)

**Comment by steve-todorov** [comment]
@johanhaleby ping :)

**Comment by johanhaleby** [comment]
@steve-todorov Sorry for the very late response but thanks for the PR :)

I just have a question regarding the "nio-multipart-parser" dependency, why is this necessary? I'm always hesitant to include additional transitive dependencies.

File: modules/spring-web-test-client/pom.xml [hunk 1]
```diff
@@ -82,6 +82,12 @@
             <artifactId>spring-webflux</artifactId>
             <version>${spring.version}</version>
         </dependency>
+        
+        <dependency>
+            <groupId>org.synchronoss.cloud</groupId>
+            <artifactId>nio-multipart-parser</artifactId>
```
Comment (johanhaleby) [inline modules/spring-web-test-client/pom.xml]: Why is this needed?

**Comment by johanhaleby** [comment]
It looks like it works without this extra dependency so I'm going to merge it to master

**Comment by steve-todorov** [comment]
Thanks for merging @johanhaleby! It looks like the `nio-multipart-parser` dependency is needed - the `master` is broken in Travis and on my machine as well. Adding the dependency back fixed it for me. Maybe @anki2189 can share more details if this needs to be scoped to only `test` or if it's actually needed? :)

File: modules/spring-web-test-client/pom.xml (hunk 1 above)
Comment (anki2189) [reply modules/spring-web-test-client/pom.xml]: It was required for webflux to support multipart file upload.

**Comment by steve-todorov** [comment]
By the way, do you plan to cut a release soon? We have a few tests which required this feature and had to disable them until this gets resolved.
